price_feeds_poller.py

```diff
@@ -316,25 +322,34 @@ def log_loop(
             contract = wpf_contract(w3, contractAddr)
             pf["contract"] = contract
             if contractAddr != "0x0000000000000000000000000000000000000000":
-              try:
-                pf["auto_disabled"] = False
-                pf["cooldown"] = int(pfs_config['feeds'][pf['caption']].get("minSecsBetweenUpdates", 0))
-                pf["deviation"] = pfs_config['feeds'][pf['caption']].get("deviationPercentage", 0.0)
-                pf["heartbeat"] = int(pfs_config['feeds'][pf['caption']].get("maxSecsBetweenUpdates", 0))                
-                pf["isRouted"] = pfs_config['feeds'][pf['caption']].get("isRouted", False)
-                if pf["isRouted"] == False:
-                  pf["witnet"] = contract.functions.witnet().call()
-                pf["lastPrice"] = int(contract.functions.lastPrice().call())
-                pf["lastRevertedTx"] = ""
-                pf["lastTimestamp"] = contract.functions.lastTimestamp().call()
-                pf["latestQueryId"] = contract.functions.latestQueryId().call()
-                pf["pendingUpdate"] = contract.functions.pendingUpdate().call()
-                pf["reverts"] = 0
-                pf["fees"].clear()
-                pf["secs"].clear()
-
-              except Exception as ex:
-                print(f"{caption} >< unable to read metadata from new contract {contractAddr}: {ex}")
+              for attempt in range(5):
+                try:
+                  # read from config
+                  pf["cooldown"] = int(pfs_config['feeds'][pf['caption']].get("minSecsBetweenUpdates", 0))
+                  pf["deviation"] = pfs_config['feeds'][pf['caption']].get("deviationPercentage", 0.0)
+                  pf["heartbeat"] = int(pfs_config['feeds'][pf['caption']].get("maxSecsBetweenUpdates", 0))
+                  pf["isRouted"] = pfs_config['feeds'][pf['caption']].get("isRouted", False)
+                  
+                  # read from web3
+                  if pf["isRouted"] == False:
+                    pf["witnet"] = contract.functions.witnet().call()
+                  pf["lastPrice"] = int(contract.functions.lastPrice().call())
+                  pf["lastTimestamp"] = contract.functions.lastTimestamp().call()
+                  pf["latestQueryId"] = contract.functions.latestQueryId().call()
+                  pf["pendingUpdate"] = contract.functions.pendingUpdate().call()
+
+                  # reset flags
+                  pf["fees"].clear()
+                  pf["secs"].clear()
+                  pf["auto_disabled"] = False
+                  pf["lastRevertedTx"] = ""
+                  pf["reverts"] = 0
+                  break
+                except Exception as ex:
+                  if attempt >= 4:
+                    print(f"{caption} >< unable to read metadata from new contract {contractAddr}: {ex}")
+                  else:
+                    continue
```

Comment:
Is this missing a `raise` or `exit(1)`? Or it is ok to keep running this function in case of error? I recommend using the same logic as in the for loop above, to avoid confusion. Something like:

```suggestion
                  if attempt < 4:
                    print(f"<{caption}>< Reading metadata, attempt #{attempt}: {ex}")
                    continue
                  else:
                    raise
```

Reply:
Also a short sleep between attempts would be nice:

```
time.sleep(1)
```